code/espurna/config/sensors.h

```diff
@@ -1255,9 +1268,13 @@
 // ADC
 // -----------------------------------------------------------------------------
 
+// In esp.h ADC_VCC and ADC_TOUT is not macro, so define macro.
+#define ADC_VCC_VALUE                   255
+#define ADC_TOUT_VALUE                  33
```

Comment:
Why hard-code it here to proxy ADC_VCC & ADC_TOUT? enum value == int when used in C-style definition.
We can keep using Esp.h enum directly

Reply:
But I think we can't define macro with value from enum.
We need this values on pre-compiler level cause it used in #if directives for define ADC_MODE_VALUE.
We also can try to #define ADC VCC and ADCT OUT but I'm afraid it's broke compatibility with esp.h.

Reply:
We can define anything as token, it is just text.

I am not seeing any preprocessor checks with ADC_MODE_VALUE, ADC_VCC_VALUE, ADC_TOUT_VALUE. So what I mean is:
```cpp
// sensor.h
#ifndef ADC_MODE_VALUE
#define ADC_MODE_VALUE ADC_VCC
#endif
```
(and slightly below when we force MODE_VALUE to become TOUT with analog sensors
```cpp
// sensor.h
#undef ADC_MODE_VALUE
#define ADC_MODE_VALUE ADC_TOUT
```

Reply:
If we define ADC_MODE_VALUE as text "ADC_VCC" we can't compare it's value with ADC_VCC value defined in esp.h. We often use ADC_VCC in utils.ino, for example.

Reply:
We can't compare things in preprocessor, true, but the there is no issue with:
```cpp
if (ADC_MODE_VALUE == ADC_VCC) {
```
gcc expands MODE_VALUE to ADC_VCC essentially making this expression either `if (true) {` or `if (false) {`

Reply:
I agree, but SDK use it in utils.ino

// This method is called by the SDK to know where to connect the ADC
int __get_adc_mode() {
    return (int) (ADC_MODE_VALUE);
}

Reply:
But the function is called at runtime, preprocessor puts correct enum value here which is converted back to int.

Building analog sensor binary, objdump -d firmware.elf
```asm
402422cc <_Z14__get_adc_modev>:
402422cc:       122c                    movi.n  a2, 33 <<< here is 33 aka ADC_TOUT
402422ce:       f00d                    ret.n
```

Reply:
You right, sorry.
We need just rewrite #if to if in all ADC_ expressions.